clang/test/Driver/sycl-oneapi-gpu.cpp

```diff
@@ -0,0 +1,384 @@
+// RUN:   FileCheck %s --check-prefix=BAD_NVIDIA_INPUT
+// RUN: %clang_cl -c -fsycl -fsycl-targets=nvidia_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_NVIDIA_INPUT
+// BAD_NVIDIA_INPUT: error: SYCL target is invalid: 'nvidia_gpu_bad'
+
+/// test for invalid amd arch
+// RUN: %clangxx -c -fsycl -fsycl-targets=amd_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_AMD_INPUT
+// RUN: %clang_cl -c -fsycl -fsycl-targets=amd_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_AMD_INPUT
+// BAD_AMD_INPUT: error: SYCL target is invalid: 'amd_gpu_bad'
+
+/// Test for proper creation of fat object
+// RUN: %clangxx -c -fsycl -fsycl-targets=intel_gpu_skl \
+// RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=FATO
+// FATO: clang-offload-bundler{{.*}} "-type=o"
+// FATO: "-targets=sycl-spir64_gen-unknown-unknown-skl,host-x86_64-unknown-linux-gnu"
+
+/// Test for proper creation of fat object
+// RUN: %clangxx -c -fsycl -fsycl-targets=nvidia_gpu_sm_50 \
+// RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=NVIDIA_FATO
+// NVIDIA_FATO: clang-offload-bundler{{.*}} "-type=o"
+// NVIDIA_FATO: "-targets=sycl-nvptx64-nvidia-cuda-sm_50,host-x86_64-unknown-linux-gnu"
+
+/// Test for proper creation of fat object
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx700 \
+// RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=AMD_FATO
+// AMD_FATO: clang-offload-bundler{{.*}} "-type=o"
+// AMD_FATO: "-targets=host-x86_64-unknown-linux,hipv4-amdgcn-amd-amdhsa--gfx700"
+
+/// Test for proper consumption of fat object
+// RUN: touch %t.o
+// RUN: %clangxx -fsycl -fsycl-targets=intel_gpu_skl \
+// RUN:   -target x86_64-unknown-linux-gnu -### %t.o 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=CONSUME_FAT
+// CONSUME_FAT: clang-offload-bundler{{.*}} "-type=o"
+// CONSUME_FAT: "-targets=host-x86_64-unknown-linux-gnu,sycl-spir64_gen-unknown-unknown-skl"
+// CONSUME_FAT: "-unbundle" "-allow-missing-bundles"
+
+/// Test for proper consumption of fat object
+// RUN: touch %t.o
+// RUN: %clangxx -fsycl -fsycl-targets=nvidia_gpu_sm_50 \
+// RUN:   -target x86_64-unknown-linux-gnu -### %t.o 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=NVIDIA_CONSUME_FAT
+// NVIDIA_CONSUME_FAT: clang-offload-bundler{{.*}} "-type=o"
+// NVIDIA_CONSUME_FAT: "-targets=host-x86_64-unknown-linux-gnu,sycl-nvptx64-nvidia-cuda-sm_50"
+// NVIDIA_CONSUME_FAT: "-unbundle" "-allow-missing-bundles"
+
+/// Test for proper consumption of fat object
+// RUN: touch %t.o
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx700 \
+// RUN:   -target x86_64-unknown-linux-gnu -### %t.o 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=AMD_CONSUME_FAT
+// AMD_CONSUME_FAT: clang-offload-bundler{{.*}} "-type=o"
+// AMD_CONSUME_FAT: "-targets=host-x86_64-unknown-linux-gnu,sycl-amdgcn-amd-amdhsa-gfx700"
```

Comment:
target string here does not creation from above comment https://github.com/intel/llvm/pull/7348/files#diff-7d1cd8526aafe83193445a154eb4721812c51cc146f8529dbe3e2f019da9653bR225